### PR TITLE
fix(host): end epoch interruption, don't await

### PR DESCRIPTION
**Pull request by brooksmtownsend**
## Feature or Problem
This PR fixes the issue where we still had remaining references to the wasmtime Engine when the host was stopping, so the epoch interruption thread didn't complete.

This may or may not be the "correct" way to solve this problem. I approached this from the perspective that
1. This error is misleading when the host is shutting down
1. We want to make sure we finish executing the thread that's running the epoch interrupter, whether or not we've properly removed all references to the underlying engine.

## Related Issues
Fixes https://github.com/wasmCloud/wasmCloud/issues/3316

### CODE REVIEW

**Comment by brooksmtownsend** [comment]
Hold off on merging until #3313 goes in just to keep merge conflicts simple

**Review by rvolosatovs — CHANGES_REQUESTED** [review]
I don't think this is the approach we should take here. Ultimately, this disables the timer and allows all components to run forever.
Since we are in the shutdown sequence anyway, why don't we simply *not* abort the interrupt thread and let it shutdown itself? So why don't we simply remove the `epoch_end` await from original implementation?

This would avoid a memory leak if the host was initialized/stopped multiple times (or there are a few instances of it)

I do still think it's concerning that we're just aborting component execution here instead of a graceful shutdown, but I suppose we can address that later (which may break logical assumptions in components, like Go `defer`)

**Comment by brooksmtownsend** [comment]
> So why don't we simply remove the epoch_end await from original implementation?

I think this makes the most sense, it's shutting down anyways so we can just remove it

> I do still think it's concerning that we're just aborting component execution here instead of a graceful shutdown, but I suppose we can address that later (which may break logical assumptions in components, like Go defer)

Yeah I think that also makes sense, during shutdown we basically need to drain the RPC messages and handle the existing ones for a graceful shutdown. Feels like a good followup issue since we really haven't needed it yet

**Review by rvolosatovs — APPROVED** [review]

**Pull request merged**
